File: Module 8/Siblings.java

/** ***
 * Author: Glowstudent
 * Program: Siblings
 **** */
import java.util.Scanner;

public class Siblings {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int uInput;

        try {

            System.out.print("Enter number of siblings: ");
            uInput = sc.nextInt();

            if (uInput >= 0) {
                if (uInput == 0) {
                    System.out.print("Ah you are an only child.");
                } else {
                    String[] siblings = new String[uInput];

                    sc.nextLine();

                    for (int i = 0; i < uInput; i++) {
                        System.out.print("Enter sibling number " + (i + 1) + " name: ");
                        siblings[i] = sc.nextLine();
                    }

                    System.out.println("\nYour siblings are:");
                    for (int i = 0; i < uInput; i++) {
                        System.out.println(siblings[i]);
                    }
                }
            } else {
                System.out.println("Invalid entry for number of siblings! Program is exiting!");
            }

        } catch (Exception ignored) {
            System.out.println("Invalid entry for number of siblings! Program is exiting!");
        }

        sc.close();
    }
}

File: Module 8/Siblings.cpp
/*****
 * Author: Glowstudent
 * Program: Siblings
 *****/

#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main()
{
    int uInput;

    cout << "Enter number of siblings: ";
    cin >> uInput;

    if (cin.fail() || uInput < 0)
    {
        cout << "Invalid entry for number of siblings! Program is exiting!" << endl;
        return 1;
    }

    if (uInput == 0)
    {
        cout << "Ah you are the only child." << endl;
    }
    else
    {
        vector<string> siblings(uInput);
        cin.ignore();
        for (int i = 0; i < uInput; i++)
        {
            cout << "Enter sibling number " << i + 1 << " name: ";
            getline(cin, siblings[i]);
        }

        cout << "\nYour siblings are:\n";
        for (int i = 0; i < uInput; i++)
        {
            cout << siblings[i] << endl;
        }
    }
    return 0;
}